x-pack/plugins/observability_solution/infra/server/routes/metadata/lib/get_metric_metadata.ts

```diff
@@ -79,17 +95,25 @@ export const getMetricMetadata = async (
     {
       metrics?: InfraMetadataAggregationResponse;
       nodeName?: InfraMetadataAggregationResponse;
+      monitoredHost?: { name: InfraMetadataAggregationResponse };
     }
   >(requestContext, 'search', metricQuery);
 
   const buckets =
     response.aggregations && response.aggregations.metrics
       ? response.aggregations.metrics.buckets
       : [];
+  const hostWithSystemIntegration =
+    response.aggregations && (response.aggregations?.monitoredHost?.name?.buckets ?? []).length > 0
+      ? response.aggregations?.monitoredHost?.name.buckets[0]?.key
+      : null;
+
+  const hasSystemIntegration = hostWithSystemIntegration === nodeId;
```

Comment:
This could also be run only when `nodeType === 'host'`

Reply:
Done 👍